### PR TITLE
RCT unpacker plus DQM 74X

### PR DESCRIPTION
**Pull request by nsmith-**
Here is a combo of the RCT unpacker + standard sequence change + DQM updates for L1TRCT and L1TdeRCT.
This PR contains the exact same unpacker as #10220, rebased to 74X.

An identical PR for 75X is #10631
A PR for the subset for 76X that hasn't been merged already
(namely standard sequence + DQM updates)
is #10658

@isobelojalvo, @jingyucms, @mcepeda, you all might be interested in this.

### CODE REVIEW

**Comment by deguio** [comment]
please test
the equivalent of this PR for the DQM part will have to be tested in 76 first. you might want to decouple DQM and unpacker if you need the latter to go in quickly.

ciao,
F.

File: Configuration/StandardSequences/python/RawToDigi_cff.py [hunk 1]
```diff
@@ -55,6 +58,7 @@
 
 RawToDigi = cms.Sequence(csctfDigis
                          +dttfDigis
+                         +l1RctHwDigis
```
Comment (nsmith-) [inline Configuration/StandardSequences/python/RawToDigi_cff.py]: It is unclear to me whether this belongs in the sequence here.

**Comment by mulhearn** [comment]
@nsmith- the name should be "rctDigis" not "rctHwDigis".  Note that in standard sequences, we always have things like:  
https://github.com/cms-sw/cmssw/blob/CMSSW_7_6_X/L1Trigger/Configuration/python/SimL1Emulator_cff.py#L26
not sure why we do it this way (predates me) but in short you don't have to worry that your unpacked digis will collide with sim digis.  I think this is a case where being different will just confuse matters...

**Comment by isobelojalvo** [comment]
@richard-cms 
"If 'rctDigis' is already in the 'process' object you might get an error,
but that means that some other version of the unpacker has incorrectly been
included in some sequence and that needs a fix."

OK so then you say @nsmith-   is right, this could cause a problem. Unfortunately, relying on others to configure their CMS SW correctly is not a very safe way to go (as we have seen over the past few years at CMS this very often causes errors on a wide scale). 

@mulhearn , we'll defer to you as the convener.

**Comment by mulhearn** [comment]
If we decide this is a lousy convention (currently used everywhere in L1T configs, since before my time) I would rather change all the names in one PR (consistently throughout L1T) instead of just making the change here to this one single module...  so for this PR, definitely stick with the convention!

**Comment by jingyucms** [comment]
Hi Nick:
can you put your change regarding 
DQM/L1TMonitor/test/l1t_dqm_sourceclient-file_cfg.py and
DQM/L1TMonitor/test/l1temulator_dqm_sourceclient-file_cfg.py

also into 
DQM/Integration/python/test/l1t_dqm_sourceclient-file_cfg.py and
DQM/Integration/python/test/l1temulator_dqm_sourceclient-file_cfg.py

Thanks and Cheers
Jingyu

**Comment by isobelojalvo** [comment]
@jingyucms  I suppose yes, he can. What is the plan for releasing this at P5?

**Comment by nsmith-** [comment]
Hi @jingyucms 
I have included the changes, although it appears to me that the blocks of code in both files that switch the source to `rawDataCollector` are taken care of in the standard sequence config here:
https://github.com/cms-l1t-offline/cmssw/blob/RCTunpackerPlusDQM74X/Configuration/StandardSequences/python/RawToDigi_cff.py#L95

Either way, better safe than sorry!

**Comment by deguio** [comment]
hello @nsmith- 
was a separate PR to 76 for the DQM part submitted already?
thanks,
F.

**Comment by nsmith-** [comment]
@deguio It was forthcoming until now.
I have updated my top comment, but for reference: #10658

**Comment by nsmith-** [comment]
@mulhearn, I have made the change to the producer label to `rctDigis`, can you please re-review this PR?

**Comment by mulhearn** [comment]
please test

**Comment by mulhearn** [comment]
+1

**Comment by deguio** [comment]
-1
apart from the cout to be removed, the PR looks ok for what concerns DQM and can go in.
rejecting for now, but feel free to update this branch and the PR will be reconsidered.

keep in mind that this one will need to be merged and succeed a relval cycle in 76 before it can get merged in 74 and 75. 

ciao,
F.

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
+1

**Comment by danduggan** [comment]
-1
still needs another commit before ready to go in, same as #10631 and #10658.

**Comment by mcepeda** [comment]
Dear Dan, all,
Can we solve this issue? We do not know what  commit is missing.
 "caloStage1Digis"  is in our understanding the unpacked digis from stage1. If we need to use another collection please let us know. 
We understood that the stage1 dqm had taken care of producing these digis already - and they must have had this exact problem with the RelVal, since in 2012 there was no Stage1 FED. 
We can try to work around the issue (even if we do not understand why this is a problem?), but we would like to understand what did the Stage1 DQM do to prevent the RelVal failing.
Any help is very welcom, this is actually very urgent.  This PR should be in the DQM online when we start taking data again.
Regards,
Maria

**Comment by danduggan** [comment]
Hi @mcepeda, the failures mean that there's something not protected in the code when running the dqm offline step, so even if it's expected to work for future runs, it would be good that it doesn't crash when these digis aren't present. There's an ongoing discussion on this in  #10658. In the meantime, if this is needed for online DQM, we can still add it on top of the release in the online DQM cluster while we converge on this. @jingyucms knows this procedure well :)

**Comment by mcepeda** [comment]
Hi @danduggan. We do need this for online DQM, urgently. @jingyucms , can you please help us with that?
Thanks!

**Comment by danduggan** [comment]
Hi @mcepeda, thanks for the confirmation. I'm starting the testing for online now (taking the rest of this off git).

**Comment by jingyucms** [comment]
@danduggan Let me testing on l1t local first.

**Comment by jingyucms** [comment]
@danduggan Can you trigger git test again?

**Comment by danduggan** [comment]
please test

**Comment by mulhearn** [comment]
+1

**Comment by nsmith-** [comment]
Superceded by #10880

**Pull request closed without merging**